components/esp-tls/esp_tls_wolfssl.c

```diff
@@ -288,6 +288,13 @@ static esp_err_t set_client_config(const char *hostname, size_t hostlen, esp_tls
             free(use_host);
             return ESP_ERR_WOLFSSL_SSL_SET_HOSTNAME_FAILED;
         }
+        /* Mimic the semantics of mbedtls_ssl_set_hostname(), which does not only set the SNI,
+           but if not set will make the mbedTLS stack skip certificate verification. */
+        if ((ret = wolfSSL_CTX_UseSNI(tls->priv_ctx, WOLFSSL_SNI_HOST_NAME, use_host, strlen(use_host))) != WOLFSSL_SUCCESS)
+        {
```

Comment:
```suggestion
        /* Mimic the semantics of mbedtls_ssl_set_hostname().  */
        if ((ret = wolfSSL_CTX_UseSNI(tls->priv_ctx, WOLFSSL_SNI_HOST_NAME, use_host, strlen(use_host))) != WOLFSSL_SUCCESS) {
```

Sorry I missed this syntax change previously.
I think commit needs to be updated as well, mbedTLS stack does not skip certificate verification.

Reply:
@frankencode Can you please make this one change. 
Thanks

Reply:
I actually copied that comment from mbedtls documentation. The documentation on [mbedtls_ssl_set_hostname](https://mbed-tls.readthedocs.io/en/latest/kb/how-to/use-sni/) states:
> Warning: If the hostname is not set with this function, Mbed TLS will silently skip certificate verification entirely. Always set the hostname with this function - even when not using SNI!

For the esp_tls_mbedtls.c logic this means if cfg->skip_common_name is true mbedtls might do as the documenation says and skip any checks on the certificate send by the server. 

But you are right, that might be an issue that needs to be raised on the mbedtls backend side and the comment is misplaced here!

Reply:
I've updated the comment. Check out the new commit.